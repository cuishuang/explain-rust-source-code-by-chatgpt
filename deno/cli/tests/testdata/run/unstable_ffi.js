const scope = import.meta.url.slice(-7) === "#worker" ? "worker" : "main";

console.log(scope, Deno.UnsafeCallback);
console.log(scope, Deno.UnsafeFnPointer);
console.log(scope, Deno.UnsafePointer);
console.log(scope, Deno.UnsafePointerView);
console.log(scope, Deno.dlopen);

if (scope === "worker") {
  postMessage("done");
} else {
  const worker = new Worker(`${import.meta.url}#worker`, { type: "module" });
  worker.onmessage = () => Deno.exit(0);
}
