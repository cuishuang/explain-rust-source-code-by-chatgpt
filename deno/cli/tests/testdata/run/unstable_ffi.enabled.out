main [class UnsafeCallback]
main [class UnsafeFnPointer]
main [class UnsafePointer]
main [class UnsafePointerView]
main [Function: dlopen]
worker [class UnsafeCallback]
worker [class UnsafeFnPointer]
worker [class UnsafePointer]
worker [class UnsafePointerView]
worker [Function: dlopen]
